### ISSUE TITLE
The retrieval-breadth instrument reports 0 URLs on every live chain, contradicting #126 — either it reads an empty channel or grounding is not happening

### ISSUE THREAD

**Issue by keiranholloway** [keiranholloway/biffo-plugin-marketing#136]
**The retrieval-breadth instrument reports zero on every live chain it has measured, and that contradicts the evidence #101 was closed on.** Until this is settled, no claim about retrieval quality — in either direction — can be trusted.

## The measurement

CloudWatch Insights over `/aws/lambda/tabsii-platform-dev-plugin-host`, 48h to 2026-08-13 06:00. **Three** breadth lines exist in that window and all three are identical:

```
research retrieval breadth: 0 distinct URLs across 2 grounded run(s)
(0 shared by more than one, 0 deep pages, 0 site roots)
```

The most recent, chain `590ec257-7422-412f-97d1-856ce1777a1d` at 05:00:55 and 05:01:03 today, with structured fields:

```json
{"research_runs_measured": 2, "research_distinct_urls": 0,
 "research_shared_urls": 0, "research_deep_pages": 0, "research_site_roots": 0}
```

The deployed code is current — the log emits from `_log_evidence_profile:951`, the post-#124 statement (the 2026-08-12 line came from `:911`, before it), and `services/marketing/src/marketing/pipeline.py` vendored into tabsii matches `dev`.

## Why this is not simply "retrieval returned nothing"

`runs_measured: 2` means two run views had `annotations is not None` — so the tri-state worked and these are **grounded runs reporting an empty list**, not absent ones. But #126, from the same period, states retrieval *"went from 4 bare homepages to 9 deep pages"*. Both cannot be true of the same pipeline.

The likely reconciliation, and the reason this needs a live check rather than more reading: **#126's figure was read off the artefact's citations; this instrument reads `AgentRunView.annotations`.** Those are different channels. `admin_app.py:440` populates it from Core's `AgentRunResponse.annotations` (biffo-template#1528/#1530).

## Two candidate causes, both serious

1. **The instrument reads a channel Core is not filling.** The plumbing exists in source — the runtime parses `message.annotations` (`agent_runtime/openrouter.py:371,414`), Core's schema carries the field (`schemas/agent_run.py:104,176`), and #1530 is merged — but I did not verify the **deployed** runtime carries #1530. If it does not, the breadth line is structurally always zero and measures nothing.
2. **Retrieval genuinely returns nothing**, and the artefact's citations are the model reproducing URLs from its own weights rather than from search — which would be considerably worse than #101's original complaint, and would mean the citation guard is passing on unretrieved sources.

## The decisive test

Run research on a real campaign, then compare, for the same `causation_id`:
- the `research retrieval breadth` line's `research_distinct_urls`, and
- the count of distinct URLs in the resulting research artefact's `citations`.

**Equal and non-zero** → everything is fine and the three zeroes were failed or degraded chains. **Artefact non-zero, instrument zero** → cause 1, and the instrument is broken. **Both zero** → cause 2, and grounding is not happening at all.

## Why this ranks above the retrieval work itself

A measuring instrument that silently reads zero is worse than no instrument: #101 was closed partly on a claim this contradicts, and the next person to ask "is retrieval any good?" gets a confident zero with no way to tell it from a real one. The `runs_measured == 0` guard added in #124 deliberately distinguishes "not measured" from "measured zero" — this case is being reported as a **measured zero**, which is precisely the false-zero that guard exists to prevent, one level further in.

## Not verified

Whether the deployed agent-runtime carries #1530; whether chain `590ec257` succeeded or failed (no other log line carries its `causation_id`, so success/failure is not determinable from logs alone).

**Comment by keiranholloway** [comment]
Claimed at 2026-08-13T06:27:28Z by `Keiran Holloway`. Release it — remove the label — on merge, or if you stop.

**Comment by keiranholloway** [comment]
Cause found, and it is **neither of the two candidates in the issue body**. It is the model family move (#108), not #120 and not a Core plumbing gap.

## The measurement

`annotations` on every research run, read from `/api/v1/admin/agent-runs/{id}`, with the model from each run's own `definition_snapshot`:

| Run (UTC) | `definition_snapshot.model` | `annotations` |
| --- | --- | --- |
| 2026-08-12 12:29 audience | `anthropic/claude-sonnet-4:online` | **5** |
| 2026-08-12 12:29 competitive | `anthropic/claude-sonnet-4:online` | **5** |
| 2026-08-12 15:21 competitive | `anthropic/claude-sonnet-5:online` | **0** |
| 2026-08-13 04:49 audience | `anthropic/claude-sonnet-5:online` | **0** |
| 2026-08-13 04:49 competitive | `anthropic/claude-sonnet-5:online` | **0** |

**PR #108** ("move every pipeline stage to the Claude 5 family") merged 14:16:49Z. **#120** merged 14:53:05Z. Both shipped in the same deploy, which is why the timing looked like #120 — I initially reached that conclusion myself and it is wrong.

The `:online` suffix is intact on both sides, so this is not a lost suffix. **`anthropic/claude-sonnet-5:online` returns no `message.annotations` where `sonnet-4:online` returned five.**

This also disposes of candidate cause 1 as written: Core *is* filling the channel, and the deployed runtime *does* carry #1530 — it demonstrably populated five annotations per run hours earlier on the same deployment path.

## What this does and does not mean for the guard

I checked, because the issue's cause 2 implies the citation guard is now passing on unretrieved sources. It is not, because **the guard never checked retrieval in the first place**:

- `NoCitationsError` fires on zero `sources` in the model's structured output. `pipeline.py`'s own docstring: *"That is a fact about what the model transcribed, not about what retrieval actually returned, and issue #82 is precisely the gap between the two."*
- `annotations` are used only to make that error's **message** accurate across three cases — not as a gate.
- `UncitedSourceError` (#22, #113) checks provenance for stages with a **closed** source set — positioning, channel plan, copy, each against its approved parent's citations. Research is the origin and has no such set.

So nothing regressed in the guard. What changed is that the one channel that could ever have evidenced research retrieval has gone quiet, so **#82's gap is now unobservable as well as unclosed**.

## What is still genuinely unsettled

Whether `sonnet-5:online` grounds at all:

- **(a)** It searches, and OpenRouter does not return annotations for that route — retrieval is fine, only observability is lost.
- **(b)** It does not search, and cited URLs come from the model's weights.

Evidence that does **not** settle it: I fetched the nine URLs cited by the 15:21 run. Three return 200; six return 403 — but those six are G2, Capterra (x2), SourceForge, GetApp and franchiselocal, all of which block non-browser clients. **403 is bot-blocking, not absence**, so this is not evidence of fabrication either way. The URLs are plausibly real, and their shape (`g2.com/products/franconnect/reviews`) is also exactly what a model could produce from memory.

Also inconclusive: the run's stored `messages` contain no injected search results and no URLs at all (3 messages, assistant content empty because the output was a tool call). That is consistent with `:online` injecting server-side without echoing into stored messages, so it proves nothing on its own.

## The decisive test

Run one research agent against a brief whose answer **cannot be in training data** — something recent, specific and checkable — and read two things: whether `annotations` returns non-empty, and whether the cited URLs contain the correct current fact. If it cites confidently and wrongly, that is (b) and it is serious.

## Recommendation

Until that test is run, treat every post-#108 research citation as **unverified provenance**, and do not close #101 or make any claim about retrieval quality in either direction — including my own earlier report of "9 deep pages", which measured the artefact, not retrieval, and which I should not have presented as evidence that retrieval improved.

If (a) holds, the mitigation is observability — a Core change to surface whatever grounding record the sonnet-5 route does return, or accepting the loss knowingly. If (b) holds, research must move back to a model whose grounding is demonstrable, and the cost of Claude 5 on that stage is not worth an evidence chain nobody can audit.

**Comment by keiranholloway** [comment]
Both halves settled. Closing.

## The cause was the model move, not #120 and not a Core plumbing gap

`annotations`, read per run from `/api/v1/admin/agent-runs/{id}` with the model taken from each run's own `definition_snapshot`:

| Run (UTC) | model | annotations |
| --- | --- | --- |
| 2026-08-12 12:29 | `claude-sonnet-4:online` | **5** |
| 2026-08-12 15:21 | `claude-sonnet-5:online` | **0** |
| 2026-08-13 04:49 | `claude-sonnet-5:online` | **0** |

**#108** ("move every pipeline stage to the Claude 5 family") merged 14:16:49Z; **#120** merged 14:53:05Z. Both shipped in one deploy, which is why the timing looked like #120 — I reached that wrong conclusion first and it is recorded here so nobody repeats it.

The decisive probe: a brief asking for the current UK Bank Rate and MPC decision date — trivially searchable, impossible from weights. Both agents returned `findings: []`. If `:online` were grounding on that route, results would have been injected.

Fixed in #141: research is back on `sonnet-4:online`, with the docstring corrected (it had asserted `:online` is a routing directive applied to any supported chat model — reasonable, and false) and the test that pinned that belief replaced by one asserting research stays on a route whose grounding has been observed.

## The instrument was sound

This issue's title asks whether the breadth instrument reads an empty channel or grounding is not happening. It was the second, and the instrument was right all along. Since the fix deployed, from `/aws/lambda/tabsii-platform-dev-plugin-host`:

```
research retrieval breadth: 9 distinct URLs across 2 grounded run(s)
  (1 shared by more than one, 8 deep pages, 1 site roots)

research retrieval breadth: 10 distinct URLs across 2 grounded run(s)
  (0 shared by more than one, 5 deep pages, 5 site roots)
```

Non-zero, with the deep-page split #101 wanted. The zeros it reported before were a true measurement of a real absence, which is exactly what an instrument should do.

## Two corrections worth keeping

- I reported "retrieval improved from 4 homepages to 9 deep pages" as evidence retrieval was working. That measured the **artefact**, on a run reporting zero annotations. It was not evidence about retrieval and I should not have offered it as such.
- I fetched the nine cited URLs to test for fabrication: three returned 200, six returned 403. Those six are G2, Capterra, SourceForge, GetApp and franchiselocal, all of which block non-browser clients. **403 is bot-blocking, not absence** — that test settled nothing in either direction, and reading it as exoneration would have been wrong.

## Still open elsewhere

`sonnet-5:online` returning no annotations is unexplained — whether OpenRouter does not support the web plugin on that route, or supports it without reporting citations, is not established here. `DEFAULT_RESEARCH_MODEL`'s docstring records the cheap test for revisiting it: run research on a fact that post-dates training and read `annotations`. Do not move that constant on the strength of a model being newer, which is precisely what #108 did.
